### ISSUE TITLE
Install script should support versioned channels

### ISSUE THREAD

**Issue by Oats87** [rancher/rke2#538]
Today, the install.sh script only supports `latest`, `stable`, and `testing` as channels. It should be able to support channels that are like `v1.18`, `v1.19`, `v1.18-testing`, `v1.18-latest`, and so forth.

**Comment by ShylajaDevadiga** [comment]
Verifying in v1.18.13-rc1+rke2r1 (3ff7cff12a1a74c991abe7628e0041e4659fbc67)

v1.18-latest is not an expected channel. v1.18 should point to latest version v1.18.12+rke2r2, but v1.18-latest does point to latest.
```
curl -sfL https://get.rke2.io | sudo INSTALL_RKE2_CHANNEL=v1.18 sh -
Rancher RKE2 Common (v1.18)                                                        375  B/s | 388  B     00:01    
Errors during downloading metadata for repository 'rancher-rke2-common-v1.18':
  - Status code: 404 for https://rpm.rancher.io/rke2/v1.18/common/centos/8/noarch/repodata/repomd.xml (IP: 104.24.112.22)
Error: Failed to download metadata for repo 'rancher-rke2-common-v1.18': Cannot download repomd.xml: Cannot download repodata/repomd.xml: All mirrors were tried
```
with Channel set to v1.18-latest 
```
rke2 -v
rke2 version v1.18.12+rke2r1 (b63f92a2b66e8ccfe2d3fae4814fa0e0579da5a4)
```

**Comment by davidnuzik** [comment]
@ShylajaDevadiga we discussed this. There is no <version> channels yet. This fails when you set a valid channel?

**Comment by davidnuzik** [comment]
@Oats87 moved to next milestone per our discussion. Still more work to do to support (TBD) versioned channels.
v1.19.5 has dev complete date of today. Please work on the issue as soon as you are able to; pr can get in but don't merge yet until post 1.18.13 release.

**Comment by ShylajaDevadiga** [comment]
Thats correct @davidnuzik. As shared above v1.18 needs a fix. Channels stable, latest, testing works as expected.

**Comment by atsai1220** [comment]
Having problem running the quick-start script but using **INSTALL_RKE2_VERSION** allows the script to pass.
Running this results in a 404 for the checksum file
```
curl -sfL https://get.rke2.io | INSTALL_RKE2_CHANNEL=latest sh -

[INFO]  finding release for channel latest
curl: (56) OpenSSL SSL_read: SSL_ERROR_SYSCALL, errno 104
[INFO]  using latest as release
[INFO]  downloading checksums at https://github.com/rancher/rke2/releases/download/latest/sha256sum-amd64.txt
curl: (22) The requested URL returned error: 404 Not Found
```

Trying that in my browser results in 404 too.
https://github.com/rancher/rke2/releases/download/latest/sha256sum-amd64.txt

**Comment by ShylajaDevadiga** [comment]
@atsai1220 I have tried using channel latest as seen below. Possibly a bug with curl as mentioned here https://github.com/curl/curl/issues/4409

Could you lease open a new issue if you still see the error and provide the OS and OS version you are using.
Ubuntu 20.04
```
curl -sfL https://get.rke2.io| sudo INSTALL_RKE2_CHANNEL=latest sh -
[INFO]  finding release for channel latest
[INFO]  using v1.18.13+rke2r1 as release
[INFO]  downloading checksums at https://github.com/rancher/rke2/releases/download/v1.18.13+rke2r1/sha256sum-amd64.txt
[INFO]  downloading tarball at https://github.com/rancher/rke2/releases/download/v1.18.13+rke2r1/rke2.linux-amd64.tar.gz
[INFO]  verifying installer
[INFO]  unpacking tarball file

rke2 -v
rke2 version v1.18.13+rke2r1
```

Validated using channel as v1.18, stable, latest, testing on CentOS

```
curl -sfL https://get.rke2.io| sudo INSTALL_RKE2_CHANNEL=v1.18 sh -
rke2 -v
rke2 version v1.18.13+rke2r1
kubectl get nodes
NAME                                          STATUS   ROLES         AGE   VERSION
ip-172-31-22-112.us-east-2.compute.internal   Ready    etcd,master   92s   v1.18.13+rke2r1
```

**Comment by atsai1220** [comment]
@ShylajaDevadiga Thank you for the update. I have created a new issue here: https://github.com/rancher/rke2/issues/623#issue-770279442
